### PR TITLE
Add theming functionality

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**

Core-UI related changes:

- copy themes' `css-variables.css` files from `@sap-theming` module to the `public` folder
- create static `css` files for each supported theme to allow customization
- create the `useTheme` hook to manage theming and communicate with Luigi
- store the theme data in localStorage
- add a theme chooser section to the `preferences` view 

Core changes:
- load the theme from localStorage on Luigi init
- handle communication from microfrontend regarding theming

**Related issue(s)**

### CODE REVIEW

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/CreateServiceBindingForm.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default function CreateServiceBindingForm({
   }, [setValidity]);
 
   useEffect(() => {
-    if (!existingInstanceName || !serviceBindings?.length) {
+    if (!existingInstanceName) {
```
Comment (akucharska) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/CreateServiceBindingForm.js:44]: Why have you removed it? If you try to create binding on brand new namespace (without any service bindings the view will crash)

File: core-ui/src/components/Preferences/ThemeChooser.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { useTheme, Dropdown } from 'react-shared';
+import { LayoutPanel } from 'fundamental-react';
+
+const THEMES = [
+  {
+    key: 'default',
+    text: 'Light (default)',
+  },
+  // {
```
Comment (akucharska) [inline core-ui/src/components/Preferences/ThemeChooser.js:10]: Can it be removed?

File: core-ui/src/components/Preferences/ThemeChooser.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+export default function ThemeChooser() {
+  const { theme, setTheme } = useTheme();
+
+  return (
+    <LayoutPanel className="fd-margin--tiny fd-margin-top--md">
+      <LayoutPanel.Header>
+        <LayoutPanel.Head title="Theme" />
+        <LayoutPanel.Actions>
+          {/* Choose the theme of the app */}
```
Comment (akucharska) [inline core-ui/src/components/Preferences/ThemeChooser.js:37]: This is a simple component so the comment is not really necessary.

**Review by akucharska — CHANGES_REQUESTED** [review]

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/CreateServiceBindingForm.js (hunk 1 above)
Comment (parostatkiem) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ServiceBindings/CreateServiceBindingForm.js:44]: something went wrong with merging from `main`. I'll revert this change

File: core-ui/src/components/Preferences/ThemeChooser.js (hunk 2 above)
Comment (parostatkiem) [reply core-ui/src/components/Preferences/ThemeChooser.js:10]: I left it on purpose for the reviewer to uncomment the code, check this theme out and confirm or deny my finding that this is identical to the `dark` theme.

File: core-ui/src/components/Preferences/ThemeChooser.js (hunk 3 above)
Comment (parostatkiem) [reply core-ui/src/components/Preferences/ThemeChooser.js:37]: Originally it was a text displayed on the page. Should I remove it or display it to the user?

File: core-ui/src/components/Preferences/ThemeChooser.js (hunk 2 above)
Comment (akucharska) [reply core-ui/src/components/Preferences/ThemeChooser.js:10]: For me it is identical as light theme. Looks like something is loading properly

File: core-ui/src/components/Preferences/ThemeChooser.js (hunk 3 above)
Comment (akucharska) [reply core-ui/src/components/Preferences/ThemeChooser.js:37]: Remove

**Review by akucharska — APPROVED** [review]

**Review by akucharska — APPROVED** [review]

**Pull request merged**